### PR TITLE
Refined SCU for Size, Position and Radius checking each state member

### PR DESCRIPTION
**Pull request by volfied**
Addresses #3554, correctly implementing `shouldComponentUpdate` for Size, Position and Radius components.

### CODE REVIEW

File: src/js/jsx/sections/transform/Position.jsx [hunk 1]
```diff
@@ -53,7 +53,12 @@ define(function (require, exports, module) {
         },
 
         shouldComponentUpdate: function (nextProps, nextState) {
-            return this.state !== nextState;
+            return !this.state ||
```
Comment (shaoshing) [inline src/js/jsx/sections/transform/Position.jsx]: Do you think `getInitialState` will handle this better? Then you don't need the state checker here. Also, if the state is never set (just guessing), the component will always re-rendered upon receiving new props.

BTW, I see other state like `scrubX` `scrubY` and they are not check in the SCU, is it intentional?

File: src/js/jsx/sections/transform/Size.jsx [hunk 2]
```diff
@@ -47,7 +47,12 @@ define(function (require, exports, module) {
         shouldComponentUpdate: function (nextProps, nextState) {
             // Calculations that are usually done here are done in
             // componentWillReceiveProps
-            return this.state !== nextState;
+            return !this.state ||
```
Comment (shaoshing) [inline src/js/jsx/sections/transform/Size.jsx]: Same question about the state existence checker.

File: src/js/jsx/sections/transform/Position.jsx (hunk 1 above)
Comment (volfied) [inline src/js/jsx/sections/transform/Position.jsx]: I'll give getInitialState a try, when I used it before, it was leaving the component unrendered, but that might have been the side effect of using `===`

`scrubX/scrubY` are states that don't affect rendering, and everytime we scrub, component receives new props to re-render. So it is intentional.

**Comment by iwehrman** [comment]
Looks great, thanks!

**Pull request merged**